extension/encoding/jaegerencodingextension/extension.go

```diff
@@ -5,27 +5,40 @@ package jaegerencodingextension // import "github.com/open-telemetry/opentelemet
 
 import (
 	"context"
+	"errors"
 	"fmt"
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/pdata/ptrace"
 )
 
 var _ ptrace.Unmarshaler = &jaegerExtension{}
+var _ ptrace.Marshaler = &jaegerExtension{}
 
 type jaegerExtension struct {
 	config      *Config
 	unmarshaler ptrace.Unmarshaler
+	marshaler   ptrace.Marshaler
 }
 
 func (e *jaegerExtension) UnmarshalTraces(buf []byte) (ptrace.Traces, error) {
 	return e.unmarshaler.UnmarshalTraces(buf)
 }
 
+func (e *jaegerExtension) MarshalTraces(td ptrace.Traces) ([]byte, error) {
```

Comment:
just don't implement it so it's not even possible to cast to Marshaler, until we know we have it

Reply:
Yeah, that's true. But I kept it as an empty function because we will eventually need to implement these [codec interfaces](https://github.com/open-telemetry/opentelemetry-collector/issues/6272#issuecomment-1554178791).

Reply:
don't introduce an empty function, do that in a separate PR. It will be nicer. And you could do it now while this PR is being reviewed.

Reply:
yes, will do it in separate PR. For now, I've removed it.